components/brave_sync/brave_sync_service_unittest.cc

```diff
@@ -778,37 +797,30 @@ TEST_F(BraveSyncServiceTest, OnSetupSyncHaveCode_Reset_SetupAgain) {
                               ContainsDeviceRecord(SyncRecord::Action::A_DELETE,
                                                    "this_device")))
       .Times(1);
+  EXPECT_CALL(*sync_client(), SendFetchSyncRecords);
   sync_service()->OnResetSync();
-  // Here we have marked service as in resetting state
+
   // Actual kSyncEnabled will go to false after receiving confirmation of
   // this_device DELETE
   EXPECT_TRUE(
       profile()->GetPrefs()->GetBoolean(brave_sync::prefs::kSyncEnabled));
 
-  EXPECT_TRUE(sync_service()->GetResettingForTest());
-
-  // OnSyncStateChanged actually is invoked 9 times:
-  // ForceCompleteReset
-  //    ResetSyncInternal
-  //        brave_sync::Prefs::Clear()
-  //            device_name
-  //            enabled => false
-  //            bookmarks_enabled
-  //            site_settings_enabled
-  //            history_enabled
-  //            device_list
-  //       enabled => false
-  // OnSetupSyncHaveCode()
-  //     device_name
-  //     enabled => true
-
-  // OnSyncEnabledChanged is actually invoked 3 times:
-  // see preference enabled in abobe list
-  EXPECT_CALL(*observer(), OnSyncStateChanged(sync_service()))
-      .Times(AtLeast(1));
-  EXPECT_CALL(*sync_client(), OnSyncEnabledChanged).Times(AtLeast(1));
-  sync_service()->OnSetupSyncHaveCode("word1 word2 word5", "test_device");
-  EXPECT_FALSE(sync_service()->GetResettingForTest());
+  // Emulating resolved with DELETE
+  records.at(0)->action = SyncRecord::Action::A_DELETE;
+  EXPECT_CALL(*sync_client(), OnSyncEnabledChanged).Times(1);
+  EXPECT_CALL(*observer(), OnSyncStateChanged(sync_service())).Times(4);
```

Comment:
can you add comment explaining what causes the StateChange for 4 times?

Reply:
addressed in https://github.com/brave/brave-core/pull/3988/commits/602ba932c83b1abda1c5b6043919c89f94b644bf